src/transactions/test/LiquidityPoolWithdrawTests.cpp

```diff
@@ -342,5 +343,93 @@ TEST_CASE("liquidity pool withdraw", "[tx][liquiditypool]")
             REQUIRE(acc1.getTrustlineBalance(poolNative1) == 0);
             checkLiquidityPool(*app, poolNative1, 0, 0, 0, 1);
         }
+
+        SECTION("large deposit/withdraw test")
+        {
+            struct Deposit
+            {
+                TestAccount acc;
+                int64_t numPoolShares;
+
+                Deposit(TestAccount const& acc, int64_t numPoolShares)
+                    : acc(acc), numPoolShares(numPoolShares)
+                {
+                }
+            };
+
+            std::vector<Deposit> deposits;
+
+            const int numAccounts = 100;
```

Comment:
Why 100?
Wouldn't a better test (with a similar runtime) be to just use 2 accounts where you test all possible combinations of pool share ratio for the subtest `InnerTest` that is `{deposit from acct1, acct2, swaps, withdraw acct2, acct1}`?
Like account1, account2 balances cover all possible (i, N-i) for i in 1..N/2 and N=50.

Reply:
Yeah this is a interesting point. The number of accounts in a pool don't matter on trades or withdrawals. I reworked the test.